subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -1583,6 +1586,26 @@ private ConfigurationIdentity getIdentity() {
         return new DefaultConfigurationIdentity(buildPath, projectPath, name);
     }
 
+    private void maybeWarnOnDeprecatedUsage(String methodName, ProperMethodUsage... properUsages) {
+        for (ProperMethodUsage properUsage : properUsages) {
+            if (properUsage.isProperUsage(this)) {
+                return;
+            }
+        }
+
+        String msgTemplate = "Calling configuration method '%s' is deprecated for configuration '%s', which has permitted usage(s):\n" +
+                "%s\n" +
+                "This method is only meant to be called on configurations which allow the (non-deprecated) usage(s): '%s'.";
+        String currentUsageDesc = UsageDescriber.describeCurrentUsage(this);
+        String properUsageDesc = ProperMethodUsage.summarizeProperUsage(properUsages);
+
+        DeprecationLogger.deprecateBehaviour(String.format(msgTemplate, methodName, getName(), currentUsageDesc, properUsageDesc))
+                .withAdvice("Configurations should only be used as permitted.")
```

Comment:
This advice doesn't really seem actionable. The advice should generally [say what to do](https://github.com/gradle/gradle/blob/master/subprojects/logging/src/main/java/org/gradle/internal/deprecation/DeprecatedFeatureUsage.java#L102-L107) about the deprecation warning. We should either move this into the deprecation message or remove it entirely

Reply:
"It hurts when I do this, doctor."  
"So don't do that anymore."

Yeah, this is redundant, I'll remove it.